permissions/actions.js

```diff
@@ -0,0 +1,83 @@
+const diff = require('deep-diff');
+
+const CLOSE_CASE = 'closeCase';
+const REOPEN_CASE = 'reopenCase';
+const ADD_NOTE = 'addNote';
+const EDIT_NOTE = 'editNote';
+const ADD_REFERRAL = 'addReferral';
+const EDIT_REFERRAL = 'editReferral';
+const ADD_HOUSEHOLD = 'addHousehold';
+const EDIT_HOUSEHOLD = 'editHousehold';
+const ADD_PERPETRATOR = 'addPerpetrator';
+const EDIT_PERPETRATOR = 'editPerpetrator';
+const ADD_INCIDENT = 'addIncident';
+const EDIT_INCIDENT = 'editIncident';
+const EDIT_CASE_SUMMARY = 'editCaseSummary';
+
+const isPathEqual = (left, right) => left[0] === right[0] && left[1] === right[1];
+
+const isAddOrEditKind = kind => ['N', 'E', 'A'].includes(kind);
```

Comment:
What do these letters stand for? (N, E, A)

Reply:
Nevermind, I've found the meaning of these letters 😄 it's in deep-diff documentation:

```
Differences are reported as one or more change records. Change records have the following structure:
kind - indicates the kind of change; will be one of the following:
N - indicates a newly added property/element
D - indicates a property/element was deleted
E - indicates a property/element was edited
A - indicates a change occurred within an array
```

Reply:
This is very specific to `deep-diff` lib:
```
kind - indicates the kind of change; will be one of the following:
N - indicates a newly added property/element
D - indicates a property/element was deleted
E - indicates a property/element was edited
A - indicates a change occurred within an array
```
 I'll make sure to update the code to make it more readable.